admin/bootstrap.py

```diff
@@ -218,6 +218,40 @@ def install_pip_dependencies(
     """
     Install Python dependencies via pip into virtualenv.
     """
+
+    # Ansible version 2.9.* cannot be directly upgraded and must be removed
+    # before attempting to install a later version - so let's check for it
+    # and uninstall it if we find it!
+
+    ansible_vercheck_cmd = [
+        os.path.join(VENV_DIR, "bin", "python3"),
+        "-c",
+        "from importlib.metadata import version as v; print(v('ansible'))",
+    ]
+
+    ansible_uninstall_cmd = [
+        os.path.join(VENV_DIR, "bin", "pip3"),
+        "uninstall",
+        "-y",
+        "ansible",
+    ]
+
+    ansible_ver = subprocess.run(
+        maybe_torify() + ansible_vercheck_cmd, text=True, capture_output=True
+    )
+    if re.match(r"^2.9+", ansible_ver.stdout):
```

Comment:
Did you mean to match a literal `.` and forget to escape it? Couldn't this be `ansible_ver.stdout.startswith("2.9")`?

Reply:
yup, good catch!